### PR TITLE
Updated TS and auto-TS to collect orch abrt saisdkdump files

### PR DESCRIPTION
**Pull request by vivekrnv**
Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

#### What I did

- Update auto-techsupport to wait for lock file creation which signifies that the saisdkdump is collected by the host before stopping syncd.
- Enhance techsupport:
   1) Add an ability the collect the saisdkdump files (if present under /var/log/orch_abrt_sdkdump/)
   2) collect_mellanox is updated to check if the syncd container status and create_switch call is successful (by checking if the PortInitDone is populated in APPL_DB) before invoking saisdkdump under syncd.

#### How to verify it

**Simulate a SAI programming failure**
```
root@r-panther-13:/home/admin# sonic-db-cli STATE_DB SET ORCH_ABRT_STATUS 1
root@r-panther-13:/home/admin# docker exec -it swss kill -6 47  (SIGABRT orchagent, which creates a coredump and triggers auto-ts)
```

**LOGS**
```
Nov 17 01:34:07.837066 r-panther-13 INFO systemd[1]: Stopping syncd service...
Nov 17 01:42:49.083737 r-panther-13 NOTICE root: sai_sdk_dump_1668642162.tar.gz collected before stopping syncd

Nov 17 01:42:50.304222 r-panther-13 INFO coredump_gen_handler.py[233582]: Waited until the saisdkdump is collected, proceeding forward..
Nov 17 01:44:18.572194 r-panther-13 INFO coredump_gen_handler.py[233582]: show techsupport --silent --global-timeout 60 --since 2 days ago is successful, sonic_dump_r-panther-13_20221117_014251.tar.gz is created
```
 
**DUMP**
```
root@r-panther-13:/home/admin/sonic_dump_r-panther-13_20221117_014251# ls -l orch_abrt_sdkdump/
total 5148
-rw-r--r-- 1 root root 1756648 Nov 17 01:44 sai_sdk_dump_1668638456.tar.gz -> ../log/sai_sdk_dump_1668638456.tar.gz
-rw-r--r-- 1 root root 1756954 Nov 17 01:44 sai_sdk_dump_1668641647.tar.gz -> ../log/sai_sdk_dump_1668641647.tar.gz
-rw-r--r-- 1 root root 1753301 Nov 17 01:44 sai_sdk_dump_1668642162.tar.gz -> ../log/sai_sdk_dump_1668642162.tar.gz
```

### CODE REVIEW

File: scripts/generate_dump [hunk 1]
```diff
@@ -879,33 +879,47 @@ enable_logrotate() {
...
         for file in $(find /tmp/sdk-dumps -type f); do
             save_file ${file} sai_sdk_dump false
         done
         rm -rf /tmp/sdk-dumps
+    else
+        # Check if the SAI_DUMP_STORE_PATH is accessible on the host and collect files from there
+        if [[ -d $sdk_dump_path ]]; then
+            $CP -rvf $sdk_dump_path /tmp/sdk-dumps
+            for file in $(find /tmp/sdk-dumps -type f); do
+                save_file ${file} sai_sdk_dump false
```
Comment (dgsudharsan) [inline scripts/generate_dump:919]: We may need to remove this logic as now techsupport will contain the same saisdkdumps in two paths one under /var/log and another under sai_sdk_dump

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: scripts/generate_dump (hunk 1 above)
Comment (vivekrnv) [reply scripts/generate_dump:919]: We collect new dumps under `/var/log/orch_abrt_status`, fwdump me collects under SAI_DUMP_STORE_PATH i.e. $sdk_dump_path. So, they won't be same.

Comment (dgsudharsan) [reply scripts/generate_dump:919]: No. What I meant was there will be two paths due to save_file being called.
By default everything under /var/log will be collected
The save_file will create a new directory and copy the dump file which means techsupport will have same dump file under two paths.
1) Remove this save_file logic or
2) After doing save file, delete the file under /var/log

Comment (vivekrnv) [reply scripts/generate_dump:919]: Only individual files under /var/log are collected by default. https://github.com/vivekrnv/sonic-utilities/blob/4bce0208da1b6f9c7cfbd23c5f931f7b8bc1b41b/scripts/generate_dump#L838 So, there won't be two paths

Comment (vivekrnv) [reply scripts/generate_dump]: Fixed. Please review

File: scripts/generate_dump [hunk 2]
```diff
@@ -866,6 +866,52 @@ enable_logrotate() {
...
+    local end_t=0
+    local filename=$1
+    local dest_dir=$2
+    local src_dir=$3
+    local do_tar_append=${4:-true}
+    local file_basename=$(basename $filename)
+    local tar_path="$BASE/$dest_dir/$file_basename"
+
+    if [[ ! -d "$TARDIR/$dest_dir" ]]; then
+        $MKDIR $V -p "$TARDIR/$dest_dir"
```
Comment (dgsudharsan) [inline scripts/generate_dump:899]: No need if check when using mkdir -p

**Review by dgsudharsan — CHANGES_REQUESTED** [review]

File: scripts/generate_dump (hunk 2 above)
Comment (vivekrnv) [reply scripts/generate_dump]: Updated

**Review by dgsudharsan — APPROVED** [review]

**Pull request closed without merging**